nodepool/config/templates/stack-template.json

```diff
@@ -8,8 +8,50 @@
         "TargetCapacity": {{$.Worker.SpotFleet.TargetCapacity}},
         "SpotPrice": "{{$.Worker.SpotFleet.SpotPrice}}",
         "LaunchSpecifications": [
+          {{if .Worker.TopologyPrivate}}
+          {{range $subnetIndex, $subnet := $.Worker.PrivateSubnets}}
```

Comment:
I'm not sure if I've understood exactly what  you meant but I thought the general idea was to get rid of that many if/else blocks introduced by this private subnet patch.
I hope you like the solution I've applied and I will extend it to the controllers and etcd sections if you approve it